v1/tester/reporter.go

```diff
@@ -57,44 +57,50 @@ func (r PrettyReporter) Report(ch chan *Result) error {
 	}
 
 	if fail > 0 && (r.Verbose || r.FailureLine) {
-		fmt.Fprintln(r.Output, "FAILURES")
+		_, _ = fmt.Fprintln(r.Output, "FAILURES")
 		r.hl()
 
 		for _, failure := range failures {
-			fmt.Fprintln(r.Output, failure)
-			if r.Verbose {
-				fmt.Fprintln(r.Output)
-				topdown.PrettyTraceWithOpts(newIndentingWriter(r.Output), failure.Trace, topdown.PrettyTraceOptions{
-					Locations:     true,
-					ExprVariables: r.LocalVars,
-				})
-			}
+			_, _ = fmt.Fprint(r.Output, failure.string(false))
+			_, _ = fmt.Fprintln(r.Output)
+
+			if len(failure.SubResults) > 0 {
+				// Print trace collectively for all sub-results.
+				if err := printFailure(r.Output, failure.Trace, r.Verbose, false, r.LocalVars); err != nil {
+					return err
+				}
+
+				if r.Verbose || r.FailureLine {
+					_, _ = fmt.Fprintln(r.Output)
+				}
+
+				for fullName, sr := range failure.SubResults.Iter {
+					w := newIndentingWriter(r.Output)
 
-			if r.FailureLine {
-				fmt.Fprintln(r.Output)
-				for i := len(failure.Trace) - 1; i >= 0; i-- {
-					e := failure.Trace[i]
-					if e.Op == topdown.FailOp && e.Location != nil && e.QueryID != 0 {
-						if expr, isExpr := e.Node.(*ast.Expr); isExpr {
-							if _, isEvery := expr.Terms.(*ast.Every); isEvery {
-								// We're interested in the failing expression inside the every body.
-								continue
+					if sr.Fail {
+						if len(sr.SubResults) == 0 {
+							// Print full test-case lineage for every failed leaf sub-result for readability.
+							for _, n := range fullName {
+								_, _ = fmt.Fprintf(w, "%s: %s\n", n, sr.outcome())
+								w = newIndentingWriter(w)
+							}
+
+							if err := printFailure(w, sr.Trace, false, r.FailureLine, r.LocalVars); err != nil {
+								return err
 							}
 						}
-						_, _ = fmt.Fprintf(newIndentingWriter(r.Output), "%s:%d:\n", e.Location.File, e.Location.Row)
-						if err := topdown.PrettyEvent(newIndentingWriter(r.Output, 4), e, topdown.PrettyEventOpts{PrettyVars: r.LocalVars}); err != nil {
-							return err
-						}
-						_, _ = fmt.Fprintln(r.Output)
-						break
 					}
 				}
+			} else {
+				if err := printFailure(r.Output, failure.Trace, r.Verbose, r.FailureLine, r.LocalVars); err != nil {
+					return err
+				}
 			}
 
-			fmt.Fprintln(r.Output)
+			_, _ = fmt.Fprintln(r.Output)
```

Comment:
All these println's ignoring errors (as they should) and aways passing the same output... perhaps it would be cleaner to just have Println method on the reporter which returned nothing and always used r.Output